packages/compass-connections/src/hooks/use-connection-repository.ts

```diff
@@ -0,0 +1,144 @@
+import { usePreference } from 'compass-preferences-model/provider';
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import ConnectionString from 'mongodb-connection-string-url';
+import { merge } from 'lodash';
+import isEqual from 'lodash/isEqual';
+import { ConnectionStorageEvents } from '@mongodb-js/connection-storage/renderer';
+import { useConnectionStorageContext } from '@mongodb-js/connection-storage/provider';
+import { useState, useEffect, useCallback } from 'react';
+import { BSON } from 'bson';
+
+type DeepPartial<T> = T extends object
+  ? { [P in keyof T]?: DeepPartial<T[P]> }
+  : T;
+
+export type PartialConnectionInfo = DeepPartial<ConnectionInfo> &
+  Pick<ConnectionInfo, 'id'>;
+
+/**
+ * Same as _.isEqual but taking into consideration BSON values.
+ */
+export function areConnectionsEqual(
+  listA: ConnectionInfo[],
+  listB: ConnectionInfo[]
+): boolean {
+  return isEqual(
+    listA.map((a: any) => BSON.serialize(a)),
+    listB.map((b: any) => BSON.serialize(b))
+  );
+}
+
+function ensureWellFormedConnectionString(connectionString: string) {
+  new ConnectionString(connectionString);
+}
+
+function sortedAlphabetically(a: ConnectionInfo, b: ConnectionInfo): number {
+  const aName = a.favorite?.name?.toLocaleLowerCase() || '';
+  const bName = b.favorite?.name?.toLocaleLowerCase() || '';
+  return aName.localeCompare(bName);
+}
+
+export type ConnectionRepository = {
+  favoriteConnections: ConnectionInfo[];
+  nonFavoriteConnections: ConnectionInfo[];
+  saveConnection: (info: PartialConnectionInfo) => Promise<ConnectionInfo>;
+  deleteConnection: (info: ConnectionInfo) => Promise<void>;
+};
+
+export function useConnectionRepository(): ConnectionRepository {
+  const storage = useConnectionStorageContext();
+
+  const [favoriteConnections, setFavoriteConnections] = useState<
+    ConnectionInfo[]
+  >([]);
+
+  const [nonFavoriteConnections, setNonFavoriteConnections] = useState<
+    ConnectionInfo[]
+  >([]);
+
+  const persistOIDCTokens = usePreference('persistOIDCTokens');
```

Comment:
Because we do different logic depending of it being activated or not. If this flag is on, we will save the tokens in the storage. On line 119:

```js
if (!persistOIDCTokens) {
        if (infoToSave.connectionOptions.oidc?.serializedState) {
          infoToSave.connectionOptions.oidc.serializedState = undefined;
        }
      }
```

Reply:
Ahh because at the end the save is from the connectionRepository hook which reflects that behaviour. Neat.